File: lib/src/main/java/uk/iatom/rfm69/registers/values/Osc1.java
package uk.iatom.rfm69.registers.values; import uk.iatom.rfm69.registers.IRegisterValue;

public enum Osc1 implements IRegisterValue {
    
    // RegOsc1),
    RCCAL_START(0x80),
    RCCAL_DONE(0x40);

    private byte val;
    
    Osc1(int val) {
        this.val = (byte) val;
    }

    @Override public byte val() {
        return val;
    }
}

File: lib/src/main/java/uk/iatom/rfm69/register/values/Osc1.kt
package uk.iatom.rfm69.register.values

import uk.iatom.rfm69.register.IRegisterValue

enum class Osc1(`val`: Int): IRegisterValue { // RegOsc1),
    RCCAL_START(0x80),
    RCCAL_DONE(0x40);

    private val `val`: Byte

    init {
        this.`val` = `val`.toByte()
    }

    override fun `val`(): Byte {
        return `val`
    }
}
